### PR TITLE
SOLR-16337: implement Zk metrics

### PR DESCRIPTION
**Pull request by noblepaul**
implement metrics for ZK ops

### How to use?

- start your servers 
- hit the end point `http://localhost:8983/solr/admin/metrics?key=solr.node:CONTAINER.zkClient`

sample output
```
{
  "metrics":{
    "solr.node:CONTAINER.zkClient":{
      "watchesFired":29,
      "reads":452,
      "writes":171,
      "bytesRead":1260447,
      "bytesWritten":483353,
      "multiOps":26,
      "cumulativeMultiOps":34,
      "childFetches":139,
      "cumulativeChildrenFetched":388,
      "existsChecks":427,
      "deletes":3}}}
```

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/metrics/SolrMetricsIntegrationTest.java [hunk 1]
```diff
@@ -216,4 +223,65 @@ public void testCoreContainerMetrics() {
     Gauge<?> g = (Gauge<?>) metrics.get("CONTAINER.fs.path");
     assertEquals(g.getValue(), cc.getSolrHome());
   }
+
+  @Test
+  public void testZkMetrics() throws Exception {
+    System.setProperty("metricsEnabled", "true");
+    MiniSolrCloudCluster cluster =
+        new MiniSolrCloudCluster.Builder(3, createTempDir())
+            .withJettyConfig(jetty -> jetty.enableV2(true))
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/metrics/SolrMetricsIntegrationTest.java:232]: isn't this enabled by default?

File: solr/core/src/test/org/apache/solr/metrics/SolrMetricsIntegrationTest.java [hunk 2]
```diff
@@ -216,4 +223,65 @@ public void testCoreContainerMetrics() {
...
+      String url = j.getBaseUrl() + "/admin/metrics?key=solr.node:CONTAINER.zkClient&wt=json";
+      HttpClient httpClient = ((HttpSolrClient) j.newClient()).getHttpClient();
+      @SuppressWarnings("unchecked")
+      Map<String, Object> zkMmetrics =
+          (Map<String, Object>)
+              Utils.getObjectByPath(
+                  Utils.executeGET(httpClient, url, Utils.JSONCONSUMER),
+                  false,
+                  List.of("metrics", "solr.node:CONTAINER.zkClient"));
+      System.out.println(Utils.toJSONString(zkMmetrics));
```
[the remark this answers is not in the document]
Comment (noblepaul) [reply solr/core/src/test/org/apache/solr/metrics/SolrMetricsIntegrationTest.java:247]: sorry, missed out

File: solr/core/src/test/org/apache/solr/metrics/SolrMetricsIntegrationTest.java (hunk 1 above)
Comment (noblepaul) [reply solr/core/src/test/org/apache/solr/metrics/SolrMetricsIntegrationTest.java:232]: yes, this is enabled by default now

File: solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java [hunk 3]
```diff
@@ -632,11 +685,17 @@ public Stat setData(String path, Path data, boolean retryOnConnLoss)
...
       throws InterruptedException, KeeperException {
+    List<OpResult> result = null;
     if (retryOnConnLoss) {
-      return zkCmdExecutor.retryOperation(() -> keeper.multi(ops));
+      result = zkCmdExecutor.retryOperation(() -> keeper.multi(ops));
     } else {
-      return keeper.multi(ops);
+      result = keeper.multi(ops);
+    }
+    metrics.multiOps.increment();
```
[the remark this answers is not in the document]
Comment (noblepaul) [reply solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java:694]: yeah, this is just counting once for a batch of operation. I'm not sure if we should make this more expensive (yet) by reading each individual operation

**Comment by noblepaul** [comment]
#986 is opened after merging with main

**Pull request closed without merging**